gamification-crowdin-services/src/main/java/io/meeds/gamification/crowdin/plugin/SuggestionApprovedTriggerPlugin.java

```diff
@@ -34,20 +32,45 @@ public void initData() {
         crowdinTriggerService.addPlugin(this);
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public List<Event> getEvents(String trigger, Map<String, Object> payload) {
+    public List<Event> getEvents(String trigger, Map<String, Object> payload, Object object) {
         if (extractSubItem(payload, getPayloadObjectName(), "provider") != null) {
             LOG.warn("Crowdin event {} translation provider is TM or MT", EVENT_TRIGGER);
             return Collections.emptyList();
         }
 
-        return Collections.singletonList(new Event(SUGGESTION_APPROVED_EVENT_TITLE,
-                null,
+        List<Event> eventList = new ArrayList<>();
+        eventList.add(new Event(APPROVE_SUGGESTION_EVENT_TITLE,
+                extractSubItem(payload, getPayloadObjectName(), "user", "username"),
                 extractSubItem(payload, getPayloadObjectName(), "user", "username"),
                 extractSubItem(payload, getPayloadObjectName(), "id"),
                 EVENT_PAYLOAD_OBJECT_NAME,
                 extractSubItem(payload, getPayloadObjectName(), "string", "project", "id"),
                 trigger.equals(CANCELLING_EVENT_TRIGGER)));
+
+        if (object == null) {
+            return eventList;
+        }
+
+        Map<String, Object> translationAuthorsMap = (Map<String, Object>) object;
+        Map<Long, String> translationsAuthors = (Map<Long, String>) translationAuthorsMap.get("translationIdUsernameMap");
+        String translationId = extractSubItem(payload, getPayloadObjectName(), "id");
+        if (translationId != null) {
+            String authorUsername = translationsAuthors.get(Long.parseLong(translationId));
+            LOG.info("authorUsername: "  + authorUsername);
```

Comment:
debug maybe ?

Reply:
this map will grow in memory, in order to keep an eye maybe we could log.debug its size instead ?

Reply:
Should have removed it. Not necessary.

Reply:
> log.debug() maybe ?

Should have removed it! Not necessary!



> this map will grow in memory, in order to keep an eye maybe we could log.debug its size instead ?

Sure. Do you want to keep an eye on it?